File: Data Structures & Algorithms/duplicate-integer/submission-2.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        hash_map = {}
        for v in nums:
            hash_map[v] = hash_map.get(v, 0) + 1

            if hash_map[v] > 1:
                return True
        
        return False

File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hash_map = new HashMap<Integer, Integer>();

        for (int i = 0; i < nums.length;i++){
            int diff = target - nums[i];

            if (hash_map.containsKey(diff)){
                return new int[]{hash_map.get(diff), i};
            }

            hash_map.put(nums[i], i);
        }

        return new int[] {};
    }
}
